### DISCUSSION TITLE
Laravel HTTP Client VCR (record and reply HTTP requests in tests)

### DISCUSSION THREAD

**Discussion by MatanYadaev** [laravel/framework#47743] [Ideas]
Example: https://github.com/php-vcr/php-vcr

**Comment by henzeb** [comment]
There is already an api for 'recording' in the http client in laravel. If we want to do more than it does now, I suggest building upon that, and not vcr which is fun, a 50's device's name to describe  software from the last decade...

**Reply by MatanYadaev** [reply]
@henzeb Recording itself it's not enough without a way to serialize it, store it, and replay it. I also think the replay functionality should be inside the existing `Http` facade.

**Reply by joshmakar** [reply]
I really wish there were a way to use VCR with Laravel's Http class.
